.github/workflows/cross-arch-build.yml

```diff
@@ -0,0 +1,128 @@
+name: Cross-Architecture Build Test
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - release/**
+
+concurrency:
+  cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.event_name == 'push' && github.sha || github.ref }}
+
+jobs:
+  linux-amd64:
+    name: "Linux AMD64"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v5
+        with:
+          submodules: true
```

Comment:
* Are there any submodules for this to need to be `true`?
* While at it, also set fetch-depth for faster checkout?
   * Ditto for other job.